torch/fx/graph.py

```diff
@@ -137,7 +137,15 @@ def node_copy(self, node: Node, arg_transform: Callable[[Node], Argument] = lamb
             # Placeholder names are user-visible, so they should be copied as-is without normalizing them.
             name = node.name
         else:
-            name = self._name(node.name)
+            sanitized_name = node.name
+            if '_' in node.name:
+                base, maybe_idx = node.name.rsplit('_', 1)
+                try:
+                    int(maybe_idx)
+                    sanitized_name = base
+                except:
+                    pass
+            name = self._name(sanitized_name)
```

Comment:
I don't think it's fully sufficient for crazy cases when user passes `add_1_1` first (it's added without the index) and then they pass `add_1` (or `add` twice). You should tests the names with indices in the used dict too (inside `_name`), just to be safe.

Reply:
Still, my comment stands: you need to add both `self._used_names[op]` and `send._used_names[f'{op}_{i}']` inside `_register_name_used` to avoid (unlikely) situation when they first add `foo_1_1` and then `foo_1`